src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/ApprovedConsentOrderAboutToSubmitHandler.java

```diff
@@ -132,4 +133,25 @@ private Boolean isPensionDocumentsEmpty(Map<String, Object> caseData) {
         List<CaseDocument> pensionDocumentsData = documentHelper.getPensionDocumentsData(caseData);
         return pensionDocumentsData.isEmpty();
     }
+
+    private static LocalDate getApprovalDate(Map<String, Object> caseData) {
+        return Optional.ofNullable(caseData.get(CONTESTED_ORDER_DIRECTION_DATE))
+            .filter(value -> value instanceof LocalDate || value instanceof String)
+            .map(value -> {
+                if (value instanceof LocalDate) {
+                    return (LocalDate) value;
+                } else {
+                    try {
+                        return LocalDate.parse((String) value);
+                    } catch (Exception e) {
+                        log.info("Invalid Approved date of order for key '" + caseData + "': " + e.getMessage());
```

Comment:
It looks like you are outputting the whole caseData map in the log message. Is this a good idea? Same comment for line 153